app/services/spree_adyen/gateways/add_allowed_origin.rb

```
@@ -0,0 +1,39 @@
+
+        response = gateway.add_allowed_origin(allowed_origin)
+
+        if response.success?
+          log("added to gateway #{gateway.id}")
+          record.update!(adyen_allowed_origin_id: response.authorization, adyen_allowed_origin_url: allowed_origin)
+        elsif response.message['errorCode'] == ALREADY_EXISTS_ERROR_CODE
+          log('already exists', :warn)
+        else
+          Rails.error.unexpected('Cannot create allowed origin', context: { url: allowed_origin, gateway_id: gateway.id }, source: 'spree_adyen')
+        end
+      end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `call` is too high. [<1, 22, 6> 22.83/17]

(Metrics/AbcSize)